app/models/lookup_value.rb

```diff
@@ -75,7 +75,8 @@ def ensure_fqdn_exists
     md = ensure_matcher(/fqdn=(.*)/)
     return md if md == true || md == false
     fqdn = md[1].split(LookupKey::KEY_DELM)[0]
-    return true if Host.unscoped.find_by_name(fqdn) || host_or_hostgroup.try(:new_record?)
+    return true if Host.unscoped.find_by_name(fqdn) || host_or_hostgroup.try(:new_record?) ||
+        (host_or_hostgroup.present? && host_or_hostgroup.type_changed? && host_or_hostgroup.type == "Host::Managed")
```

Comment:
`host_or_hostgroup` is the discovered host itself, you can either create a lookup_value matcher yourself or override a parameter from the host or hostgroup in which can host_or_hostgroup will be the host/hostgroup that created an override which triggers a creation of the matcher.